tests/topotests/pim_boundary_acl/test_pim_boundary_acl.py

```diff
@@ -0,0 +1,525 @@
+    r1 = tgen.gears["r1"]
+
+    # No IGMP sources other than from self for AutoRP Discovery group initially
+    expected = {
+        "r1-eth0":{
+            "name":"r1-eth0",
+            "224.0.1.40":"*",
+            "229.1.1.1":None,
+            "232.1.1.1":None
+        },
+        "r1-eth2":{
+            "name":"r1-eth2",
+            "224.0.1.40":"*",
+            "229.1.1.1":None,
+            "232.1.1.1":None
+        }
+    }
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip igmp sources json", {}
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=20, wait=1)
+    assert result is None, "Expected no IGMP sources other than from AutoRP Discovery"
+
+    # Send IGMP join from r2, check if r1 has IGMP source
+    r2.vtysh_cmd((
+        """
+          configure terminal
+            interface r2-eth0
+              ip igmp join {} 10.0.20.2
+        """
+    ).format(SSM_GROUP))
+    expected = {
+        "r1-eth0":{
+            "name":"r1-eth0",
+            "232.1.1.1":{
+                "group":"232.1.1.1",
+                "sources":[
+                    {
+                        "source":"10.0.20.2",
+                        "timer":"*",
+                        "forwarded":False,
+                        "uptime":"*"
+                    }
+                ]
+            }
+        }
+    }
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip igmp sources json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=20, wait=1)
+    assert result is None, "Expected IGMP source to be present but is absent"
+
+    # Test inbound boundary on r1
+    # Enable multicast boundary on r1, toggle IGMP join on r2
+    r2.vtysh_cmd((
+        """
+          configure terminal
+            interface r2-eth0
+              no ip igmp join {} 10.0.20.2
+        """
+    ).format(SSM_GROUP))
+    r1.vtysh_cmd(
+        """
+          configure terminal
+            interface r1-eth0
+              ip multicast boundary pim-acl
+        """
+    )
+    r2.vtysh_cmd((
+        """
+          configure terminal
+            interface r2-eth0
+              ip igmp join {} 10.0.20.2
+        """
+    ).format(SSM_GROUP))
+    expected = {
+        "r1-eth0":{
+            "name":"r1-eth0",
+            "232.1.1.1":None
+        }
+    }
+    test_func = partial(
+        topotest.router_json_cmp, r1, "show ip igmp sources json", expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=20, wait=1)
+    assert result is None, "Expected IGMP source to be absent but is present"
+
+    # Add lower, more-specific permit rule to access-list
+    r2.vtysh_cmd((
+        """
+          configure terminal
+            interface r2-eth0
+              no ip igmp join {} 10.0.20.2
+        """
+    ).format(SSM_GROUP))
+    tgen.routers()["r2"].cmd(("vtysh -c 'conf t' -c 'int {}' -c 'no ip igmp join {} 10.0.20.2'").format("r2-eth0", SSM_GROUP))
```

Comment:
Any reason you don't use here as previously `r2.vtysh_cmd()`?

Reply:
Had converted to `vtysh_cmd()` but missed removing this equivalent, fixed now.